### PR TITLE
use "COPY" instead of "ADD" in Dockerfile

### LINKED ISSUE #2155: Please use "COPY" instead of "ADD" in Dockerfile
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy

Below Dockerfile has used "ADD" instead of "COPY"
./images/cockroachdb/Dockerfile:ADD rook /usr/local/bin/
./images/minio/Dockerfile:ADD rook /usr/local/bin/
./images/cross/Dockerfile:ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH} /tini
./images/ceph/Dockerfile:ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH} /tini
./images/ceph/Dockerfile:ADD rook rookflex /usr/local/bin/

**Expected behavior:**
Please use "COPY"

**How to reproduce it (minimal and precise):**

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)):

### PR DESCRIPTION
**Pull request by MartinForReal**
**Description of your changes:**
 use "COPY" instead of "ADD" in Dockerfile to adopt docker best practice

**Which issue is resolved by this Pull Request:**
Resolves #2155 

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

### CODE REVIEW

File: images/ceph/Dockerfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ ARG ARCH
 ARG TINI_VERSION
 
 # Run tini as PID 1 and avoid signal handling issues
-ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH} /tini
+RUN curl -SL -o /tini https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH}
 RUN chmod +x /tini
```
Comment (galexrt) [inline images/ceph/Dockerfile]: Make this and the above `RUN` into one RUN statement instead of two `RUN`s/layers.

File: images/cross/Dockerfile [hunk 2]
```diff
@@ -49,7 +49,7 @@ ARG ARCH
 ARG TINI_VERSION
 
 # Run tini as PID 1 and avoid signal handling issues
-ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH} /tini
+RUN curl -SL -o /tini https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH}
 RUN chmod +x /tini
```
Comment (galexrt) [inline images/cross/Dockerfile]: Same as above

**Comment by MartinForReal** [comment]
fixed as requested

File: images/ceph/Dockerfile [hunk 3]
```diff
@@ -23,8 +23,7 @@ ARG ARCH
 ARG TINI_VERSION
 
 # Run tini as PID 1 and avoid signal handling issues
-ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH} /tini
-RUN chmod +x /tini
+RUN curl -SL -o /tini https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH} && chmod +x /tini
```
Comment (galexrt) [inline images/ceph/Dockerfile]: Move the chmod to it's own line:
```
RUN curl ... && \
    chmod ...
```

File: images/cross/Dockerfile [hunk 4]
```diff
@@ -49,8 +49,7 @@ ARG ARCH
 ARG TINI_VERSION
 
 # Run tini as PID 1 and avoid signal handling issues
-ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH} /tini
-RUN chmod +x /tini
+RUN curl -SL -o /tini https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH} && chmod +x /tini
```
Comment (galexrt) [inline images/cross/Dockerfile]: Same here.

File: images/ceph/Dockerfile [hunk 5]
```diff
@@ -23,8 +23,8 @@ ARG ARCH
 ARG TINI_VERSION
 
 # Run tini as PID 1 and avoid signal handling issues
-ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH} /tini
-RUN chmod +x /tini
+RUN curl -SL -o /tini https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH} && \
```
Comment (galexrt) [inline images/ceph/Dockerfile]: I just ran the curl command to test it and we should supress the curl progress output and add the `--fail` flag.
`curl -sSL --fail -o /tini URL`

**Pull request merged**